File: java-wrapper/src/main/java/com/absmartly/wrapper/DummyContextDataProvider.java
package com.absmartly.wrapper;

import com.absmartly.sdk.ContextDataProvider;
import com.absmartly.sdk.json.ContextData;
import java8.util.concurrent.CompletableFuture;

public class DummyContextDataProvider implements ContextDataProvider {
    private ContextData nextData;

    public DummyContextDataProvider() {
        this.nextData = null;
    }

    public void setNextData(ContextData data) {
        this.nextData = data;
    }

    @Override
    public CompletableFuture<ContextData> getContextData() {
        if (nextData != null) {
            ContextData data = nextData;
            nextData = null;
            return CompletableFuture.completedFuture(data);
        }
        return CompletableFuture.completedFuture(new ContextData());
    }
}

File: kotlin-wrapper/src/main/kotlin/com/absmartly/wrapper/DummyContextDataProvider.kt
package com.absmartly.wrapper

import com.absmartly.sdk.ContextData
import com.absmartly.sdk.ContextDataProvider
import java.util.concurrent.CompletableFuture
import java.util.concurrent.atomic.AtomicReference

class DummyContextDataProvider : ContextDataProvider {
    private val nextData = AtomicReference<ContextData?>(null)

    fun setNextData(data: ContextData) {
        nextData.set(data)
    }

    override fun getContextData(): CompletableFuture<ContextData> {
        val data = nextData.getAndSet(null)
        return CompletableFuture.completedFuture(data ?: ContextData())
    }
}
